spec/dor/rights_metadata_spec.rb

```diff
@@ -50,19 +50,73 @@
+        ),
+        'metadata_source_ssi'         => 'DOR',
+        'title_tesim'                 => ['Indianapolis 500'],
         'rights_characteristics_ssim' => include(
-          'world_discover', 'has_group_rights', 'has_rule', 'group|stanford', 'world_read', 'world|no-download', 'profile:group1|world1'
+          'world_discover', 'has_group_rights', 'has_rule', 'group|stanford', 'location', 'location_with_rule',
+          'world_read', 'world|no-download', 'profile:group1|location1|world1', 'none_read_file'
         ),
-        'use_license_machine_ssi' => 'by-nc'
+        'use_license_machine_ssi'     => 'by-nc',
+        'obj_rights_locations_ssim'   => ['reading_room']
+      )
+      expect(doc).not_to include(
+        'rights_errors_ssim', 'file_rights_locations_ssim', 'obj_rights_agents_ssim', 'file_rights_agents_ssim'
+      )  # don't include empties
+    end
+
+    it 'should filter access_restricted from what gets aggregated into rights_descriptions_ssim' do
+      rights_md_ds = Dor::RightsMetadataDS.new
+      mock_dra_obj = double(Dor::RightsAuth)
+      expect(mock_dra_obj).to receive(:index_elements).with(no_args).at_least(:once).and_return(
+        :primary => 'access_restricted',
+        :errors  => [],
+        :terms   => [],
+        :obj_locations_qualified => [{:location => 'someplace', :rule => 'somerule'}],
+        :file_groups_qualified   => [{:group => 'somegroup', :rule => 'someotherrule'}]
+      )
+      expect(rights_md_ds).to receive(:dra_object).and_return(mock_dra_obj)
+
+      doc = rights_md_ds.to_solr
+      expect(doc).to match a_hash_including(
+        'rights_primary_ssi'       => 'access_restricted',
+        'rights_descriptions_ssim' => include('location: someplace (somerule)', 'somegroup (file) (someotherrule)'),
+      )
+      expect(doc).not_to match a_hash_including(
+        'rights_descriptions_ssim' => include('access_restricted')
+      )
+    end
+
+    it 'should include the simple fields that are present' do
+      rights_md_ds = Dor::RightsMetadataDS.new
+      mock_dra_obj = double(Dor::RightsAuth)
+      expect(mock_dra_obj).to receive(:index_elements).with(no_args).at_least(:once).and_return(
+        :primary => 'access_restricted',
+        :errors  => [],
+        :terms   => [],
+        :obj_locations  => ['location'],
+        :file_locations => ['file_specific_location'],
+        :obj_agents     => ['agent'],
+        :file_agents    => ['file_specific_agent']
+      )
+      expect(rights_md_ds).to receive(:dra_object).and_return(mock_dra_obj)
+
+      doc = rights_md_ds.to_solr
+      expect(doc).to match a_hash_including(
+        'obj_rights_locations_ssim'  => ['location'],
+        'file_rights_locations_ssim' => ['file_specific_location'],
+        'obj_rights_agents_ssim'     => ['agent'],
+        'file_rights_agents_ssim'    => ['file_specific_agent']
       )
-      expect(doc).not_to include('rights_errors_ssim')  # don't include empties
```

Comment:
well, that expectation was expanded to encompass the new fields also (see line 73 in the new code).  though there isn't a specific test for empties in general.  so... i think this keeps the spirit of the existing test (in that it tests the empty field behavior, though not as thoroughly as it could, since the old code could've also tested empty behavior for `rights_characteristics_ssim`).

i think this is adequate (no worse than it was before), but i'm happy to break that out into a separate (and more thorough) test if you think it's worth it.